validation/manifest.go

```diff
@@ -16,7 +16,7 @@ import (
 var (
 	serviceNameValidationRegex = regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`)
 	envVarValidatorRegex       = regexp.MustCompile(`^[-._a-zA-Z][-._a-zA-Z0-9]*$`)
-	hostnameRegex              = regexp.MustCompile(`^[[:alnum:],-,\.]+\.[[:alpha:]]{2,}$`)
+	hostnameRegex              = regexp.MustCompile(`^[[:alnum:],\-,\.]+\.[[:alpha:]]{2,}$`)
```

Comment:
This is strange - I think it shouldn't need to be escaped inside of a `[]`?  `envVarValidatorRegex` doesn't seem to need to be.

Anyways, I think that we should use the validation that comes from k8s for this stuff:

* https://github.com/kubernetes/apimachinery/blob/8daf28983e6ecf28bd8271925ee135c1179ad13a/pkg/util/validation/validation.go#L208
* https://github.com/kubernetes/apimachinery/blob/8daf28983e6ecf28bd8271925ee135c1179ad13a/pkg/util/validation/validation.go#L408

Reply:
Done.